CMake/AbseilHelpers.cmake

```diff
@@ -126,15 +142,28 @@ function(absl_cc_library)
     else()
       # Generating header-only library
       add_library(${_NAME} INTERFACE)
-      target_include_directories(${_NAME}
-        INTERFACE ${ABSL_COMMON_INCLUDE_DIRS})
+      target_include_directories(${_NAME} INTERFACE
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+        $<INSTALL_INTERFACE:include>  # <prefix>/include
+      )
       target_link_libraries(${_NAME}
         INTERFACE ${ABSL_CC_LIB_DEPS} ${ABSL_CC_LIB_LINKOPTS}
       )
       target_compile_definitions(${_NAME} INTERFACE ${ABSL_CC_LIB_DEFINES})
     endif()
 
     add_library(absl::${ABSL_CC_LIB_NAME} ALIAS ${_NAME})
+
+    # install rules
+    if(ABSL_ENABLE_INSTALL)
+      install(TARGETS ${_NAME}
+        EXPORT ${PROJECT_NAME}Targets
+        INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
```

Comment:
Remove line 161, as it will result in overspecifying the `INTERFACE_INCLUDE_DIRECTORIES` property of the exported targets. For example, CMake 3.12.1 generates the following exported target for `absl::base` (in `${CMAKE_INSTALL_PREFIX}/lib64/cmake/absl/abslTargets.cmake`), if the same include directory is added twice via the `install` and `target_include_directories` commands:

```cmake
# Create imported target absl::base
add_library(absl::base STATIC IMPORTED)

set_target_properties(absl::base PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "absl::base_internal;absl::config;absl::core_headers;absl::dynamic_annotations;absl::spinlock_wait"
)
```